core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -142,6 +142,12 @@ private TableProperties() {
   // This only applies to files written after this property is set. Files previously written aren't
   // relocated to reflect this parameter.
   // If not set, defaults to a "data" folder underneath the root path of the table.
+  public static final String WRITE_FOLDER_STORAGE_LOCATION = "write.folder-storage.path";
+
+  /**
+   * @deprecated will be removed in 0.14.0, use {@link #WRITE_FOLDER_STORAGE_LOCATION} instead
+   */
+  @Deprecated
```

Comment:
+1 to the deprecation.

When the deprecation comes around, we should consider providing an action to migrate tables off of deprecated properties. In general, I think that would allow us to make smarter decisions about table property names as new needs surface. Something similar to the v2 table migration action. But that’s a long ways off.

Reply:
I think migration is unnecessary. The actual value stored is always `write.folder-storage.path`, we are just changing the Java variable name. And as I noted, we will deprecate after 2 releases, so that people can have time to change their variable name.

Reply:
Why will be removed in `0.14.0`? Is there a convention to follow?